packages/relay/src/lib/precheck.ts

```diff
@@ -376,4 +380,17 @@ export class Precheck {
       throw predefined.UNSUPPORTED_TRANSACTION_TYPE;
     }
   }
+
+  /**
+   * Checks if the receiver account exists.
+   * @param {Transaction} tx - The transaction.
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   */
+  async receiverAccount(tx: Transaction, requestDetails: RequestDetails) {
+    const verifyAccount = await this.mirrorNodeClient.getAccount(tx.to!, requestDetails);
+
+    if (verifyAccount !== null && verifyAccount.receiver_sig_required === true) {
+      throw predefined.INTERNAL_ERROR("Receiver's signature is required.");
```

Comment:
Let's throw a `RESOURCE_NOT_FOUND` predefined error instead of `INTERNAL_ERROR`. Also let's include the value of tx.to.

```suggestion
      throw predefined.RESOURCE_NOT_FOUND(`Receiver's signature is required: tx.to: ${tx.to}`);
```

Reply:
yeah, there may be an existing error that could be used but if not add a new one